src/realm.js

```diff
@@ -1485,9 +1486,9 @@ export class Realm {
 
     // materialize leaked objects
     if (leakedObjects !== undefined && leakedObjects.size !== 0) {
-      for (let ioVal of leakedObjects) {
-        invariant(ioVal !== undefined);
-        ioVal.object.materializeInGenerator(ioVal.generator);
+      for (let lo of leakedObjects) {
+        invariant(lo !== undefined);
```

Comment:
Best practice for invariants is to explain why you believe they can never be false. In this instance, I cannot verify this unless I leave the diff view (of this commit) and start reading the overall code. A comment would have saved me a bunch of time.

As it is, I can't quite figure out how lo can ever become undefined. So my question is: Why is LeakedObjects defined to allow undefined elements? In particular, what is the use case and which test case exercises that use case?

Reply:
I believe this was just a rename from `ioVal` which was the same invariant as before. So maybe @NTillmann can explain more why `leakedObjects` is a union with `void`. We should probably refine that type so that there is no union with `void`?